posthog/hogql/test/test_modifiers.py

```diff
@@ -238,3 +244,26 @@ def test_modifiers_materialization_mode(self):
             "SELECT replaceRegexpAll(nullIf(nullIf(JSONExtractRaw(events.properties, %(hogql_val_0)s), ''), 'null'), '^\"|\"$', '') AS `$browser` FROM events"
             in response.clickhouse
         )
+
+    def test_optimize_joined_filters(self):
+        # no optimizations
+        response = execute_hogql_query(
+            f"select event from events where person.properties.$browser ilike '%Chrome%'",
+            team=self.team,
+            modifiers=HogQLQueryModifiers(optimizeJoinedFilters=OptimizeJoinedFilters.false),
+        )
+        # "ilike" shows up once in the response
+        assert response is not None
+        assert response.clickhouse is not None
+        assert response.clickhouse.count("ilike") == 1
+
+        # with optimizations
+        response = execute_hogql_query(
+            f"select event from events where person.properties.$browser ilike '%Chrome%'",
+            team=self.team,
+            modifiers=HogQLQueryModifiers(optimizeJoinedFilters=OptimizeJoinedFilters.true),
+        )
```

Comment:
I played with the inlining locally, and while this test case demonstrates it working, the show stops when the `{filters}` placeholder is used (which I ran into because SQL insight have it default):

```sql
select event from event
where {filters} and person.properties.$browser ilike '%Chrome%'
```

Can we avoid quitting optimizing in this case?

Reply:
Weird, don't know what I was seeing, but it does work 🤔